Bump tracing-appender version to resolve reolution issue

btab opened the pull request:
Bump tracing-appender version to resolve "use of undeclared crate or module `tracing_subscriber`" from, for example...

tracing-appender-0.2.2/src/non_blocking.rs:59:5

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

nipunn1313 commented:
Where are you seeing this issue? When I run `cargo build` - everything builds fine. The version of `tracing-appender` and `tracing-subscriber` should be locked in the `Cargo.lock`.

btab commented:
Apologies for the delayed response.

I followed the instructions [here](https://stack.convex.dev/building-the-oss-backend).

The commit hash I cloned was `372b67b25065dc21dc00af3984764187c40e0cc3`.

OS = MacOS 14.6.1

When I ran `just run-local-backend`, it bombed out as it tried to compile tracing appender (as above). Explicitly constraining that dep to `0.2.3` fixed the issue.

nipunn1313 commented:
I still can't repro

```
cargo --version
cargo 1.79.0-nightly (c93926759 2024-04-23)
```
^ this version is pinned in the `rust-toolchain` file and is necessary for the build.

```
git rev-parse HEAD
372b67b25065dc21dc00af3984764187c40e0cc3

just run-local-backend
cargo run -p local_backend --bin convex-local-backend -- "$@"
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.35s
     Running `target/debug/convex-local-backend`
```

No issue with bumping the dep in general, but I am wondering if there's some different issue you're running into. The dep should be fine. The Cargo.lock locks all the deps - our CI is passing, it doesn't repro locally.

I am very curious about what may be different about your setup.

btab commented:
Yeah genuinely curious. For completeness (I've reverted my change locally to squirrel out this problem)...

```
% rustup toolchain list
stable-aarch64-apple-darwin (default)
nightly-2024-04-27-aarch64-apple-darwin (override)
nightly-aarch64-apple-darwin
```

```
% cargo --version
cargo 1.79.0-nightly (c93926759 2024-04-23)
```

```
% cargo check          
warning: cmd_util v0.1.0 (/Users/andre/src/convex-backend/crates/cmd_util) ignoring invalid dependency `tracing-subscriber` which is missing a lib target
    Checking reqwest v0.12.7
    Checking tonic v0.12.1
   Compiling tonic-build v0.12.1
   Compiling sqlx-macros-core v0.8.0
    Checking oauth2 v4.4.2
    Checking tracing-appender v0.2.2
   Compiling h2 v0.4.5
   Compiling sentry-core v0.32.3
   Compiling tower v0.4.13
   Compiling bindgen v0.69.4
error[E0433]: failed to resolve: use of undeclared crate or module `tracing_subscriber`
  --> /Users/andre/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-appender-0.2.2/src/non_blocking.rs:59:5
   |
59 | use tracing_subscriber::fmt::MakeWriter;
   |     ^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `tracing_subscriber`

error[E0433]: failed to resolve: use of undeclared crate or module `tracing_subscriber`
   --> /Users/andre/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-appender-0.2.2/src/rolling.rs:183:10
    |
183 | impl<'a> tracing_subscriber::fmt::writer::MakeWriter<'a> for RollingFileAppender {
    |          ^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `tracing_subscriber`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `tracing-appender` (lib) due to 2 previous errors
```

That `cmd_util` warning at the beginning might also be indiciative of the same problem?

1. From my reset repo, I tried just bumping `tracing-subscriber` in the workspace `Cargo.toml` to `0.3.18` and that cleared the error (`cargo check` passed).
2. Then I looked at a diff in `Cargo.lock` and saw that two things had changed `tracing-subscriber` was `0.3.18` as expected **and** `tracing-log` went from `0.1.3` to `0.2.0`.
3. Then I changed `tracing-subscriber` back to `=0.3.17` in `Cargo.toml`.
4. `cargo check` was back in its failure state.
5. `Cargo.lock`'s only diff at this point was `tracing-log` at `0.1.4`.

Hopefully some of that info sparks some thoughts.

nipunn1313 commented:
I'm also on 14.16.1 mac - and `cargo check` passes for me. Super curious.

I wonder if something is corrupt in your registry. The error itself that you are seeing makes no sense either. `tracing_subscriber` is clearly in the Cargo.toml of tracing-appender https://github.com/tokio-rs/tracing/blob/b2cd503f8104db8da013d54572337f2600838992/tracing-appender/Cargo.toml

You could test this hypothesis by deleting /Users/andre/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-appender-0.2.2 and then rerunning.

btab commented:
Good news: I managed to get it working without the feature bump. Though your suggestion didn't quite do the trick, it inspired me to try a more aggressive version of the same idea. I had to delete several versions of `tracing-subscriber` from several different subdirs of `.cargo`. I guess cargo's a little more fragile than I realized. I'll close the PR.
